packages/flutter/lib/src/widgets/sliver_fill.dart

```diff
@@ -41,13 +43,26 @@ class SliverFillViewport extends StatelessWidget {
   /// the viewport in the main axis.
   final double viewportFraction;
 
+  /// Whether to add padding to both ends of the list.
+  ///
+  /// If this is set to true and [viewportFraction] < 1.0, padding will be added
+  /// such that the first and last child slivers will be in the center of
+  /// the viewport when scrolled all the way to the start or end, respectively.
+  /// You may want to set this to false if this [SliverFillViewport] is not the only
+  /// widget along this main axis, such as in a [CustomScrollView] with multiple
+  /// children.
+  ///
+  /// This option cannot be [null]. If [viewportFraction] >= 1.0, no padding will be
+  /// added regardless, so this option does nothing. Defaults to [true].
+  final bool padEnds;
+
   /// {@macro flutter.widgets.sliverMultiBoxAdaptor.delegate}
   final SliverChildDelegate delegate;
 
   @override
   Widget build(BuildContext context) {
     return _SliverFractionalPadding(
-      viewportFraction: (1 - viewportFraction).clamp(0, 1) / 2,
+      viewportFraction: padEnds ? (1 - viewportFraction).clamp(0, 1) / 2 : 0,
```

Comment:
I wonder if it might be more efficient to just not include the `_SliverFractionalPadding` if `!padEnds`. Providing a padding of 0 essentially make it a no-op widget I think. WDYT @LongCatIsLooong?

Reply:
I opted not to do that initially, mostly because I'm fairly new to Flutter and I'm not 100% familiar how changing the internal structure of widgets impacts other things down the line. If we can be reasonably sure it won't affect anything, I'd definitely prefer to omit the `_SliverFractionalPadding` entirely

Reply:
Sounds good! My only concern is if `SliverFillViewport` has a stateful child, toggling `padEnds` may cause the stateful child to lose state. But that doesn't sound like a valid use case.

Reply:
I concur! :)
The potential state issue could be resolved by adding a getter/setter for `padEnds` that then marks the widget to update the state.

Reply:
Correct me if I'm wrong, but since `SliverFillViewport` is a `StatelessWidget`, isn't it supposed to be invalid to mutate its state after creation? So toggling `padEnds` wouldn't even be allowed, right?

Reply:
Oh sorry by "toggle" what I really meant was to rebuild the `SliverFillViewport` with a different `padEnds` value, e.g.:
```dart
class SomeStatefulWidget extends StatefulWidget {
  bool padEnds;

  @override
  Widget build(BuildContext context) => SliverFillViewport(padEnds: padEnds);
}   
```

Reply:
I guess it depends on how it is used! I have seen a similar issue pop up with a stateless widget that was used in the composition of a stateful widget that toggled a property.

Reply:
Ah, that makes more sense. So given that there's a workaround and it's probably an uncommon use-case, I'll have it omit the `_SliverFractionalPadding`

Reply:
I think it would have been better to not change the shape of the tree when padEnds is toggled from true to false or vice versa as that may result in some state loss in stateful widgets.

Reply:
Undid that change as it seems like it caused a post-commit test to fail anyway